### PR TITLE
Add support for pools for oneof fields

### PR DESCRIPTION
**Pull request by maticz**
I have implemented the minimal changes necessary for pools to work with oneof fields because the generated code currently does not compile. If needed, I can work on improving the code, add some tests and so on but I'd be interested first in knowing whether you'd be willing to merge something like this.

### CODE REVIEW

**Comment by vmg** [comment]
Yeah for sure, sorry about that missing functionality, we just don't have any `oneof` fields in Vitess. If you can write some tests for this I'll definitely merge. :+1:

**Comment by coufalja** [comment]
@maticz Hello, we would like to make use of the pooling of `oneof` fields. Would you mind If I build on top of your work and add the tests in a separate PR?

**Comment by maticz** [comment]
Hey @coufalja, sure, go ahead. I'd be glad if this functionality was merged.

**Comment by DoubleDi** [comment]
@vmg Hi! Sorry for the ping. 
Your lib is getting quite popular and we are starting to use it. This is a critical feature for us, as we also have oneofs. Could you consider merging this ASAP?

File: features/unmarshal/unmarshal.go [hunk 1]
```diff
@@ -530,7 +530,11 @@ func (p *unmarshal) fieldItem(field *protogen.Field, fieldname string, message *
 			p.P(`if oneof, ok := m.`, fieldname, `.(*`, field.GoIdent, `); ok {`)
 			p.decodeMessage("oneof."+field.GoName, buf, field.Message)
 			p.P(`} else {`)
-			p.P(`v := &`, msgname, `{}`)
+			if p.ShouldPool(message) && p.ShouldPool(field.Message) {
+				p.P(`v := `, msgname, `FromVTPool()`)
```
Comment (DoubleDi) [inline features/unmarshal/unmarshal.go:534]: imho there is a dot missing. `.FromVTPool()`

**Comment by vmg** [comment]
@DoubleDi Hi! I'm afraid that this PR doesn't have tests, and since we don't use OneOf fields in Vitess, I cannot properly verify that the behavior is accurate. If you're interested in seeing this PR merged, writing tests for it would be a great first step. Thank you!

**Comment by vmg** [comment]
@Fumesover has kindly written tests for the feature in https://github.com/planetscale/vtprotobuf/pull/83 -- this is now implemented.

**Pull request closed without merging**